src/alchemlyb/parsing/amber.py

```diff
@@ -223,15 +223,13 @@ def file_validation(outfile):
             )
             raise ValueError(f"no free energy section found in file {outfile}")
 
-        mbar_ndata = 0
         have_mbar, bar_intervall, mbar_states = secp.extract_section(
             "^FEP MBAR options:",
             "^$",
             ["ifmbar", "bar_intervall", "mbar_states"],
             "^---",
         )
         if have_mbar:
-            mbar_ndata = int(nstlim / bar_intervall)
```

Comment:
I assume removing these was fine and you did that on purpose. tests are passing so ... lgtm

Reply:
Yes, I had a look at this and it does seems that mbar_ndata is not used.
The reason that I want to add linting check is to prevent this from being merged in the first place.